filebeat/module/mongodb/log/_meta/fields.yml

```diff
@@ -0,0 +1,27 @@
+- name: log
+  type: group
+  description: >
+      Contains fields from MongoDB logs.
+  fields:
+  - name: timestamp
```

Comment:
Should we use `@timestamp` for this?

Reply:
What do you mean?

Reply:
Defining `timestamp` here means it will be `mongodb.log.timestamp` in the event. But normally the processed timestamp from the log line when the event happened ends up in `@timestamp`. Is this a different timestamp here?

Reply:
I think a separate `timestamp` field is required, because it stores the actual time the log was generated. `@timestamp` is the time it was processed. These two times are not necessarily the same.

Reply:
We have similar behaviour in other modules, for example Kafka. What we do there is that we have the timestamp of the actual log line in `@timestamp` and the one when the log line was actually read in `read_timestamp`. Name of `read_timestamp` is not great but we don't have a better one yet.

If you look at the ingest pipeline of the Kafka module you will see the the timestamps are moved around to make sure `@timestamp` is the one after processing.